File: Lista_8/lista8_ex10.c
/* Exercício 10:
Escreva uma função recursiva para a busca sequencial em um vetor.
*/

///BUSCA SEQUENCIAL RECURSIVA;
#include <stdio.h>
#include <stdlib.h>

int buscaSequencial(int vet[], int tam, int chave);

int main () {
int pos, chave=45, tam=10, vet[] = {20, 5, 15, 24, 67, 45, 1, 76, -10, -1};

pos = buscaSequencial(vet, tam, chave);

    if(pos != -1) {
        printf("\nA posicao da chave %d no vetor é: %d\n", chave, pos);
    } else {
        printf("\nA chave %d não está no vetor! \n", chave);
    }

return 0;
}

int buscaSequencial(int vet[], int tam, int chave) {
    if(tam == 1) {
        if(vet[0] == chave) {
            return 0;
        } else {
            return -1;
        }
    } else {
        if(vet[tam-1] == chave) {
            return tam-1;
        }
        return buscaSequencial(vet, tam-1, chave);
    }
}

File: Lista_8/Lista8_ex10.py
def busca_sequencial(vetor, tam, chave):
	if tam == 1:
		if vetor[0] == chave:
			return 0
		else:
			return -1
	else:
		if vetor[tam-1] == chave:
			return tam - 1
		return busca_sequencial(vetor, tam-1, chave)

def main():
	vetor = [20, 5, 15, 24, 67, 45, 1, 76, -10, -1]
	chave = 45

	pos = busca_sequencial(vetor, len(vetor), chave)

	if pos != -1:
		print("\nA posicao da chave {} no vetor é: {}\n".format(chave, pos))
	else:
		print("\nA chave {} não está no vetor! \n".format(chave)) 

if __name__ == "__main__":
	main()
